### PR TITLE
unique_ptr implementation

### PR DESCRIPTION
**Pull request by Aleksiiej**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
Coverage rate is less than 90%. You can download detailed coverage report in action's [summary](https://github.com/coders-school/memory-management/actions/runs/2618657651). Look for `Artifacts` in the bottom of this page

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
✅ CI Passed - Task: check_unique_ptr_interface
🏅 Granted 4 XP for @Aleksiiej

**Comment by ziobron** [comment]
Coverage rate is less than 90%. You can download detailed coverage report in action's [summary](https://github.com/coders-school/memory-management/actions/runs/2618898461). Look for `Artifacts` in the bottom of this page

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
✅ CI Passed - Task: unique_ptr_coverage
🏅 Granted 5 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
✅ CI Passed - Task: unique_ptr
🏅 Granted 10 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
✅ CI Passed - Task: shared_ptr_formatting_check
🏅 Granted 1 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: weak_ptr_students_tests
🏅 Granted 12 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task weak_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

File: homework/shared_ptr/shared_ptr.hpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            delete ptrToControlBlock_;
+        }
+    }
+}
+
+template <typename T>
+shared_ptr<T>& shared_ptr<T>::operator=(shared_ptr& other) noexcept {
+    if (&other != this && other.ptr_ != nullptr) {
+        ptr_ = other.ptr_;
+        ptrToControlBlock_ = other.ptrToControlBlock_;
```
Comment (Rogal255) [inline homework/shared_ptr/shared_ptr.hpp:122]: Potential memory leak here. When that shared_ptr was the last one then you need to delete ptr_ first before assigning new value to it. Similarly you should check if there are any weak_ptrs pointing to that control block and if not - delete it as well. The same bug can be found at least in one additional place in your code.

**Comment by ziobron** [comment]
Thanks @Rogal255 for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/177

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task check_unique_ptr_interface were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_students_tests were already granted in https://github.com/coders-school/memory-management/pull/173

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task unique_ptr_coverage were already granted in https://github.com/coders-school/memory-management/pull/174

**Comment by ziobron** [comment]
✅ CI Passed - Task: files_check
🏅 Granted 0 XP for @Aleksiiej

**Comment by ziobron** [comment]
⚠️ Points for @Aleksiiej for task shared_ptr_formatting_check were already granted in https://github.com/coders-school/memory-management/pull/174

**Pull request closed without merging**